### PR TITLE
Fixes issue #5371

### LINKED ISSUE #5371: Notice: Undefined index: file_field_file_table in file_view_file() (line 3441 of /app/core/modules/file/file.module).
## Description of the bug

Getting this notice when viewing a file (e.g. pdf) via the entity url: (file/1 for example):

> Notice: Undefined index: file_field_file_table in file_view_file() (line 3441 of /app/core/modules/file/file.module).

## Steps To Reproduce

1. Upload a file at `admin/content/files`.
2. Change the file display to something other than "Download link" at `admin/structure/file-types/manage/document/file-display`.
2. View the uploaded file by clicking on the link in the Manage files list.
3. Check logs (or display error messages to screen).

The `$display` is odd which will show something like:

```
array (
  'formatter' => 'file_field_file_default',
  'settings' => 
  array (
    'file_field_file_download_link' => 
    array (
      'text' => 'Download [file:name]',
    ),
  ),
)
```

when anything other than download link is chosen. Perhaps they need to match.

# Additional info

First reported https://github.com/backdrop/backdrop-issues/issues/5370

### PR DESCRIPTION
**Pull request by argiepiano**
Fixes [#5371](https://github.com/backdrop/backdrop-issues/issues/5371)

### CODE REVIEW

File: core/modules/file/file.admin.inc [hunk 1]
```diff
@@ -368,10 +368,13 @@ function file_display_form($form, &$form_state, $file_type, $view_mode) {
 function file_display_form_submit($form, &$form_state) {
   $file_type = $form['#file_type'];
   $view_mode = $form['#view_mode'];
+  $formatter = $form_state['values']['display']['formatter'];
+  $settings = isset($form_state['values']['display']['settings'][$formatter]) ? $form_state['values']['display']['settings']: array();
```
Comment (klonos) [inline core/modules/file/file.admin.inc]: Missing space there before the `:`:
```suggestion
  $settings = isset($form_state['values']['display']['settings'][$formatter]) ? $form_state['values']['display']['settings'] : array();
```

**Review by klonos — CHANGES_REQUESTED** [review]
LGTM, besides that missing space there, before the `:` in the shorthand conditional.

**Review by klonos — APPROVED** [review]
LGTM now 👍🏼

**Comment by quicksketch** [comment]
Looks great! Merged into 1.x and 1.20.x.

**Pull request merged**